services/meta/service_test.go

```diff
@@ -1397,3 +1447,17 @@ func mustMarshalJSON(v interface{}) string {
 	}
 	return string(b)
 }
+
+func freePort() string {
```

Comment:
I think this is simply the wrong approach period (random ports that is).  I'm going to try to refactor the entire test suite at some point to use pre-determined ports.  Circle is very problematic in how it releases and recycles random ports.